lib/ansible/plugins/lookup/mongodb.py

```diff
@@ -0,0 +1,155 @@
+        if result is None:
+            return result
+        if isinstance(result, (basestring, int, long, float, bool)):
+            return result
+        elif isinstance(result, list):
+            new_list = []
+            for elem in result:
+                new_list.append(self.convert_mongo_result_to_valid_json(elem))
+            return new_list
+        elif isinstance(result, dict):
+            new_dict = {}
+            for key, value in result.iteritems():
+                new_dict[key] = self.convert_mongo_result_to_valid_json(value)
+            return new_dict
+        elif isinstance(result, datetime.datetime):
+            #epoch
+            return (result - datetime.datetime(1970,1,1)).total_seconds()
+        else:
+            #failsafe
+            return "{}".format(result)
+
+
+    def run(self, terms, variables, **kwargs):
+
+        ret = []
+        for term in terms:
+            '''
+            Makes a MongoDB query and returns the output as a valid list of json.
+            Timestamps are converted to epoch integers/longs.
+
+            Here is a sample playbook that uses it:
+
+-------------------------------------------------------------------------------
+- hosts: all
+  gather_facts: false
+
+  vars:
+    mongodb_parameters:
+      #optional parameter, default = "mongodb://localhost/"
+      # connection_string: "mongodb://localhost/"
+
+      #mandatory parameters
+      database: 'local'
+      collection: "startup_log"
+
+      #optional query  parameters
+      #we accept any parameter from the normal mongodb query.
+      # the offical documentation is here
+      # https://api.mongodb.org/python/current/api/pymongo/collection.html?highlight=find#pymongo.collection.Collection.find
+    #   filter:  { "hostname": "batman" }
+    #   projection: { "pid": True    , "_id" : False , "hostname" : True }
+    #   skip: 0
+    #   limit: 1
+    #   sort:  [ [ "startTime" , "ASCENDING" ] , [ "age", "DESCENDING" ] ]
+
+  tasks:
+    - debug: msg="Mongo has already started with the following PID [{{ item.pid }}] - full_data {{ item }} "
+      with_items:
+      - "{{ lookup('mongodb', mongodb_parameters) }}"
+-------------------------------------------------------------------------------
+            '''
+
+            connection_string = term.get('connection_string', "mongodb://localhost")
+            database = term["database"]
+            collection =  term['collection']
+
+            if "connection_string" in term:
+                del term["connection_string"]
+            del term["database"]
+            del term["collection"]
+
+            if "sort" in term:
+                term["sort"] = self._fix_sort_parameter(term["sort"])
+
+            # all other parameters are sent to mongo, so we are future and past proof
+
+            try:
+                client = MongoClient(connection_string)
+                results = client[database][collection].find( **term )
```

Comment:
Maybe it could be a good idea to provide a way to add additional args to mongo client (timeout, sll, ...)